libc/CMakeLists.txt: `string sub-command REPLACE requires at least four arguments.`

Caused by this change: https://github.com/llvm/llvm-project/issues/125831

```
-- Targeting AArch64
CMake Error at /build/source/libc/CMakeLists.txt:54 (string):
  string sub-command REPLACE requires at least four arguments.

```
Full log:
https://llvm-jenkins.debian.net/job/llvm-toolchain-jammy-binaries/architecture=amd64,distribution=jammy,label=amd64/1939/console

--- a/libc/CMakeLists.txt
+++ b/libc/CMakeLists.txt
@@ -51,7 +51,11 @@ set(LIBC_KERNEL_HEADERS "/usr/include" CACHE STRING "Path to Linux kernel header
 # Defining a global namespace to enclose all libc functions.
 set(default_namespace "__llvm_libc")
 if(LLVM_VERSION_MAJOR)
-  string(REPLACE "-" "" NS_LLVM_VERSION_SUFFIX ${LLVM_VERSION_SUFFIX})
+  if(DEFINED LLVM_VERSION_SUFFIX AND NOT "${LLVM_VERSION_SUFFIX}" STREQUAL "")
+    string(REPLACE "-" "" NS_LLVM_VERSION_SUFFIX "${LLVM_VERSION_SUFFIX}")
+  else()
+    set(NS_LLVM_VERSION_SUFFIX "")
+  endif()
   set(default_namespace "__llvm_libc_${LLVM_VERSION_MAJOR}_${LLVM_VERSION_MINOR}_${LLVM_VERSION_PATCH}_${NS_LLVM_VERSION_SUFFIX}")
 endif()
 set(LIBC_NAMESPACE ${default_namespace}

--- a/libc/CMakeLists.txt
+++ b/libc/CMakeLists.txt
@@ -51,11 +51,7 @@ set(LIBC_KERNEL_HEADERS "/usr/include" CACHE STRING "Path to Linux kernel header
 # Defining a global namespace to enclose all libc functions.
 set(default_namespace "__llvm_libc")
 if(LLVM_VERSION_MAJOR)
-  if(DEFINED LLVM_VERSION_SUFFIX AND NOT "${LLVM_VERSION_SUFFIX}" STREQUAL "")
-    string(REPLACE "-" "" NS_LLVM_VERSION_SUFFIX "${LLVM_VERSION_SUFFIX}")
-  else()
-    set(NS_LLVM_VERSION_SUFFIX "")
-  endif()
+  string(REPLACE "-" "" NS_LLVM_VERSION_SUFFIX "${LLVM_VERSION_SUFFIX}")
   set(default_namespace "__llvm_libc_${LLVM_VERSION_MAJOR}_${LLVM_VERSION_MINOR}_${LLVM_VERSION_PATCH}_${NS_LLVM_VERSION_SUFFIX}")
 endif()
 set(LIBC_NAMESPACE ${default_namespace}

